doc/contribute/nest_with_ides.rst

```diff
@@ -0,0 +1,103 @@
+Developing NEST with IDEs
+=========================
+
+Integrated development environments can make coding more efficient and fun.
+We provide here some recipes based on practical experience on how to use
+IDEs in NEST development. The information here is not meant to be complete,
+cross-checked or up-to-date, but is provided on a best-effort basis to get
+you started. Kindly contribute your experiences.
+
+Since we generally recommend out-of-source builds, only such are discussed here.
+
+.. contents:: We currently have recipes for the following IDEs
+   :local:
+   :depth: 2
+
+Eclipse
+-------
+
+This recipe describes how to set up Eclipse for editing, building and
+running NEST. The description here was tested on macOS 11.
+
+Requirements and limitations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Focus on single build configuration
+* Assumes all dependencies (OpenMPI, GSL, etc) installed in a Conda environment
+* Does not support debugging on macOS (because Eclipse does not support lldb)
+* Does not read the NEST `.clang-format` file, so code formatting may
+  be incorrect
+* I experimented with better CMake support using ``cmake4eclipse``,
+  but did not figure it out. So instructions below rely on running
+  ``cmake`` manually in the build directory.
+* Tested with Eclipse 2020-12
```

Comment:
```suggestion
* Tested with `Eclipse 2020-12 <https://www.eclipse.org/downloads/packages/release/2020-12>`_
```

Reply:
While this link would take you to the specific version I installed, it leads to a little informative page with raw packages, which are not a good idea to install, since they do not include the Java environment. I'd suggest the link below to the Eclipse install page, even though it is not version-specific:
```suggestion
* Tested with `Eclipse 2020-12 <https://www.eclipse.org/downloads/packages/>`_.
```

Reply:
Yes, that's a good idea.